lib/productScreen.dart

```diff
@@ -16,17 +17,24 @@ class product_screen extends StatefulWidget {
 
 class _product_screenState extends State<product_screen> {
   bool isloading = true;
-  List product = [];
+  List<Product> product = [];
 
   getProduct() async {
     //to make request to get this data:
     var url = Uri.parse("https://fakestoreapi.com/products");
     var res = await http.get(url);
```

Comment:
the connection may have error, so you should wrap this code in try and catch.
@ShamsMoAhmed 
...
we will discuss the internet permission for Android manifest file.